src/hooks/screen/useChannelHook.ts

```diff
@@ -3,41 +3,45 @@ import {PostNotificationChannelList} from '../../../types/database/schema/PostNo
 
 function useChannelHook() {
   const determinePostType = (data: PostNotificationChannelList): BaseChannelItemTypeProps => {
-    const isOwningPost = data?.rawJson?.isOwnPost;
+    const isOwnSignedPost =
+      data?.rawJson?.isOwnSignedPost && data?.channelType === 'POST_NOTIFICATION';
+    const isOwnAnonymousPost =
+      data?.rawJson?.isOwnAnonymousPost && data?.channelType === 'ANON_POST_NOTIFICATION';
     const isAnonymousPost = data?.rawJson?.isAnonym;
 
     const firstComment = data?.rawJson?.comments[0];
     const hasComment = data?.rawJson?.comments.length > 0;
     const isOwningComment = firstComment?.reaction?.isOwningReaction;
     const isAnonymousComment = firstComment?.reaction?.data?.is_anonymous;
 
-    if (isOwningPost && isAnonymousPost && isOwningComment && isAnonymousComment)
+    if (isOwnAnonymousPost && isAnonymousPost && isOwningComment && isAnonymousComment)
       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_I_COMMENTED_ANONYMOUSLY;
 
-    if (isOwningPost && isAnonymousPost && isOwningComment)
+    if (isOwnAnonymousPost && isAnonymousPost && isOwningComment)
       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_I_COMMENTED;
 
-    if (isOwningPost && isAnonymousPost && isAnonymousComment)
+    if (isOwnAnonymousPost && isAnonymousPost && isAnonymousComment)
       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_COMMENTED_ANONYMOUSLY;
 
-    if (isOwningPost && isAnonymousPost && hasComment)
+    if (isOwnAnonymousPost && isAnonymousPost && hasComment)
       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_COMMENTED;
 
-    if (isOwningPost && isAnonymousPost) return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION;
+    if (isOwnAnonymousPost && isAnonymousPost)
+      return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION;
 
-    if (isOwningPost && isOwningComment && isAnonymousComment)
+    if (isOwnSignedPost && isOwningComment && isAnonymousComment)
       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_I_COMMENTED_ANONYMOUSLY;
 
-    if (isOwningPost && isOwningComment)
+    if (isOwnSignedPost && isOwningComment)
       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_COMMENTED;
 
-    if (isOwningPost && isAnonymousComment)
+    if (isOwnSignedPost && isAnonymousComment)
       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_COMMENTED_ANONYMOUSLY;
 
-    if (isOwningPost && hasComment)
+    if (isOwnSignedPost && hasComment)
       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_COMMENTED;
 
-    if (isOwningPost) return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION;
+    if (isOwnSignedPost) return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for determining the post type based on various conditions seems repetitive and could be simplified by using a lookup table or a switch-case structure. This would make the code more maintainable and easier to read.

```diff
-    if (isOwnAnonymousPost && isAnonymousPost && isOwningComment && isAnonymousComment)
-       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_I_COMMENTED_ANONYMOUSLY;
-    if (isOwnAnonymousPost && isAnonymousPost && isOwningComment)
-       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_I_COMMENTED;
-    if (isOwnAnonymousPost && isAnonymousPost && isAnonymousComment)
-       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_COMMENTED_ANONYMOUSLY;
-    if (isOwnAnonymousPost && isAnonymousPost && hasComment)
-       return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_COMMENTED;
-    if (isOwnAnonymousPost && isAnonymousPost)
-      return BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION;
-    if (isOwnSignedPost && isOwningComment && isAnonymousComment)
-       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_I_COMMENTED_ANONYMOUSLY;
-    if (isOwnSignedPost && isOwningComment)
-       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_COMMENTED;
-    if (isOwnSignedPost && isAnonymousComment)
-       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_COMMENTED_ANONYMOUSLY;
-    if (isOwnSignedPost && hasComment)
-       return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION_COMMENTED;
-    if (isOwnSignedPost) return BaseChannelItemTypeProps.MY_SIGNED_POST_NOTIFICATION;

+    const postTypeConditions = [
+      { condition: isOwnAnonymousPost && isAnonymousPost && isOwningComment && isAnonymousComment, type: BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_I_COMMENTED_ANONYMOUSLY },
+      { condition: isOwnAnonymousPost && isAnonymousPost && isOwningComment, type: BaseChannelItemTypeProps.MY_ANON_POST_NOTIFICATION_I_COMMENTED },
+      // ... add all other conditions here
+    ];
+
+    for (const { condition, type } of postTypeConditions) {
+      if (condition) return type;
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->